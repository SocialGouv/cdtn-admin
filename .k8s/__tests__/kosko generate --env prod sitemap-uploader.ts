//

import { getEnvManifests } from "@socialgouv/kosko-charts/testing";
import { project } from "@socialgouv/kosko-charts/testing/fake/gitlab-ci.env";

jest.setTimeout(1000 * 60);
test("kosko generate --prod restore", async () => {
  process.env.SOURCE_CONTAINER = "source-container";
  process.env.SOURCE_SERVER = "prod";
  process.env.DESTINATION_CONTAINER = "destination-container";
  process.env.DESTINATION_SERVER = "dev";
  expect(
    await getEnvManifests("prod", "sitemap-uploader", {
      ...project("cdtn-admin").prod,
    })
  ).toMatchSnapshot();
});
