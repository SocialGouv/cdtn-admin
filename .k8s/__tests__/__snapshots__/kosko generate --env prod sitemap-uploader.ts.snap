// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kosko generate --prod restore 1`] = `
"metadata:
  name: sitemap-uploader
  namespace: cdtn-admin-secret
spec:
  backoffLimit: 3
  template:
    metadata:
      name: sitemap-uploader
      namespace: cdtn-admin-secret
    spec:
      restartPolicy: OnFailure
      containers:
        - name: az-sitemap-uploader
          image: 'mcr.microsoft.com/azure-cli:2.9.1'
          command:
            - bash
          args:
            - '-c'
            - >+


              echo \\"Fetch sitemap from $SITEMAP_ENDPOINT\\"

              wget $SITEMAP_ENDPOINT -O sitemap.xml


              echo \\"Upload sitemap to
              azure/$DESTINATION_CONTAINER/$DESTINATION_NAME\\"

              az storage blob upload --account-name $AZ_ACCOUNT_NAME
              --account-key $AZ_ACCOUNT_KEY --container-name
              $DESTINATION_CONTAINER --file sitemap.xml --name $DESTINATION_NAME

          env:
            - name: DESTINATION_CONTAINER
              value: sitemap
            - name: DESTINATION_NAME
              value: test.xml
            - name: AZ_ACCOUNT_NAME
              value: $(azurestorageaccountname)
            - name: AZ_ACCOUNT_KEY
              value: $(azurestorageaccountkey)
            - name: SITEMAP_ENDPOINT
              value: 'https://code.travail.gouv.fr'
          envFrom:
            - secretRef:
                name: azure-cdtnadmindev-volume
apiVersion: batch/v1
kind: Job
"
`;
